src/nni_engine.hpp

```diff
@@ -68,33 +89,71 @@ class NNIEngine {
   void RunMainLoop();
   void RunPostLoop();
 
-  // ** Filter Functions
-
-  // Initialization step for filter before beginning
+  // ** Filtering Functions
+  // Initialization step for filter before beginning engine run.
```

Comment:
@ognian- can you have a chat with @davidrich27 about these definitions just below? The overhead of std::function is _definitely_ not a problem here (vs templates) but if we want to have some code standard about how to handle this let's talk.

Reply:
I think using std::function is a great option, it avoids templating and is readable.